lsfd-cmd/sock-xinfo.c

```diff
@@ -216,6 +219,42 @@ void load_sock_xinfo(struct path_cxt *pc, const char *name, ino_t netns)
 	}
 }
 
+void load_fdsk_xinfo(struct proc *proc, int fd)
+{
+	int pidfd, sk, nsfd;
+	struct netns *nsobj;
+	struct stat sb;
+
+	/* This is additional/extra information, ignoring failures. */
+	pidfd = syscall(SYS_pidfd_open, proc->pid, 0);
+	if (pidfd < 0)
+		return;
+
+	sk = syscall(SYS_pidfd_getfd, pidfd, fd, 0);
```

Comment:
Please see (and include) include/pidfd-utils.h. It seems we need to add pidfd_getfd() there.